packages/flutter_tools/lib/src/project.dart

```diff
@@ -22,6 +23,25 @@ import 'plugins.dart';
 import 'template.dart';
 import 'web/workflow.dart';
 
+FlutterProjectFactory get projectFactory => context.get<FlutterProjectFactory>() ?? FlutterProjectFactory();
+
+class FlutterProjectFactory {
```

Comment:
nit: give this a `const` constructor and make the getter above use that instead

Reply:
Done.